File: app/src/androidTest/java/com/kazakago/cleanarchitecture/di/component/TestApplicationComponent.java
package com.kazakago.cleanarchitecture.di.component;

import com.kazakago.cleanarchitecture.domain.usecase.AboutUseCaseTest;
import com.kazakago.cleanarchitecture.domain.usecase.CityUseCaseTest;
import com.kazakago.cleanarchitecture.domain.usecase.WeatherUseCaseTest;
import com.kazakago.cleanarchitecture.di.module.ApplicationModule;
import com.kazakago.cleanarchitecture.di.module.DataModule;
import com.kazakago.cleanarchitecture.di.module.DomainModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by tamura_k on 2016/06/15.
 */
@Singleton
@Component(modules = {ApplicationModule.class, DomainModule.class, DataModule.class})
public interface TestApplicationComponent {

    void inject(AboutUseCaseTest aboutUseCaseTest);

    void inject(CityUseCaseTest cityUseCaseTest);

    void inject(WeatherUseCaseTest weatherUseCaseTest);

}

File: app/src/androidTest/java/com/kazakago/cleanarchitecture/di/component/TestApplicationComponent.kt
package com.kazakago.cleanarchitecture.di.component

import com.kazakago.cleanarchitecture.di.module.ApplicationModule
import com.kazakago.cleanarchitecture.di.module.DataModule
import com.kazakago.cleanarchitecture.di.module.DomainModule
import com.kazakago.cleanarchitecture.di.module.WebModule
import com.kazakago.cleanarchitecture.domain.usecase.AppInfoUseCaseTest
import com.kazakago.cleanarchitecture.domain.usecase.CityUseCaseTest
import com.kazakago.cleanarchitecture.domain.usecase.WeatherUseCaseTest
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, DomainModule::class, DataModule::class, WebModule::class))
interface TestApplicationComponent {

    fun inject(appInfoUseCaseTest: AppInfoUseCaseTest)

    fun inject(cityUseCaseTest: CityUseCaseTest)

    fun inject(weatherUseCaseTest: WeatherUseCaseTest)

}